### PR TITLE
Bring the website (chisel5) in as git subtree

### PR DESCRIPTION
**Pull request by mwachs5**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

This brings in the website code as a single subtree, and sets up CI to run required checks that the website builds in this environment.

Note that this is a branch of the website, `chisel5`, which has already stripped out a lot of the website code.

I was able to build this locally.

```
git subtree add --prefix website git@github.com:freechipsproject/www.chisel-lang.org chisel5
```

   - code refactoring    

#### API Impact

No impact on chisel user code.

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

 @seldridge, @jackkoenig  what do you recommend? Do we want to squash all the git subtree commits into one?
I'm thinking maybe we squash this into 3 commits:
1. the existing website as all one commit (forget the git subtree fancieness as we don't expect to keep up maintenance here, and not sure how much we care about the website commit history details)
2. the changes i had to make the chisel-only version of the website
3. the changes I had to make to get it to build and run within this repo

#### Release Notes

Add the chisel-lang website code under a website subdirectory.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by mwachs5** [comment]
Do i need to make the website `all website tests passed` check required to make it run the job?

**Comment by mwachs5** [comment]
Need to make these checks actually required (by adding to the all-tests-passed list), and TBD the workflow on setting up the publishing branch

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -110,13 +110,94 @@ jobs:
...
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8
+      - name: Cache Scala
+        uses: coursier/cache-action@v5
+      - name: Calculate API Docs Cache Key
+        id: calc-key
+        run: |
+          KEY=$(make --always-make --dry-run apis-${{ matrix.project }} | sha256sum | awk '{print $1}')
```
Comment (seldridge) [inline .github/workflows/test.yml:132]: Is this supposed to be:

```suggestion
          KEY=$(make -C website --always-make --dry-run apis-${{ matrix.project }} | sha256sum | awk '{print $1}')
```

There's a failure in that CI action step which was surprising.

**Review by seldridge — APPROVED** [review]
Nice.

One question about needing to run make in the right place for the API docs.

**Comment by seldridge** [comment]
In a follow-up it would be fine to setup publishing, but to do It in this repo. We can then test that the website works under `chipsalliance.github.io/chisel3`, get changes in, then eventually switch the URL.

File: .github/workflows/test.yml (hunk 1 above)
Comment (mwachs5) [reply .github/workflows/test.yml:132]: ah, yeah this is exactly what i was worried about -- we keep hittin gin the cache so haven't tested this. I need to manually flush this cache i think to make sure it works even on a cache miss

**Comment by mwachs5** [comment]
(will clean up commit history before merging)

**Pull request merged**